From 690d2a65eae3c968e82d47f62aee1234548163a6 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Mon, 2 Oct 2023 11:29:01 +0200
Subject: [PATCH] Set GZ_TOOLS_VER to 2 for consistency with rest of Garden and
 Harmonic libraries

Signed-off-by: Silvio Traversaro <silvio.traversaro@iit.it>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a9e8ed0..7a4a6699 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,7 +97,7 @@ set(GZ_MATH_VER ${gz-math7_VERSION_MAJOR})
 # Note that CLI files are installed regardless of whether the dependency is
 # available during build time
 find_program(HAVE_GZ_TOOLS gz)
-set(GZ_TOOLS_VER 1)
+set(GZ_TOOLS_VER 2)
 
 #--------------------------------------
 # Find Tinyxml2
