From a956bc89ee8a48f04e46c380b7d2dfb1557b698f Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Wed, 4 Oct 2023 10:34:45 +0200
Subject: [PATCH] Install gz-tools commands on Windows

Signed-off-by: Silvio Traversaro <silvio.traversaro@iit.it>
---
 core/CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 2bbe9114..b2859c58 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -132,6 +132,4 @@ gz_build_tests(TYPE UNIT
 )
 
 add_subdirectory(include/gz/msgs)
-if(NOT WIN32)
-  add_subdirectory(src/cmd)
-endif()
+add_subdirectory(src/cmd)
